packages/app/components/header-dropdown.tsx

```diff
@@ -36,27 +38,51 @@ import { breakpoints } from "design-system/theme";
 type HeaderDropdownProps = {
   type: "profile" | "settings";
   withBackground?: boolean;
+  data?: UserProfile;
 };
-function HeaderDropdown({ type, withBackground = false }: HeaderDropdownProps) {
+function HeaderDropdown({
+  type,
+  withBackground = false,
+  data: user,
+}: HeaderDropdownProps) {
   const { logout } = useAuth();
   const router = useRouter();
   const { colorScheme, setColorScheme } = useColorScheme();
-  const { user } = useUser();
+  const { user: currentUser, isAuthenticated } = useUser();
   const { userAddress } = useCurrentUserAddress();
   const { width } = useWindowDimensions();
   const isWeb = Platform.OS === "web";
   const isMdWidth = width >= breakpoints["md"];
   const isDark = colorScheme === "dark";
+  const walletAddresses = useMemo(
+    () => user?.profile?.wallet_addresses_v2,
```

Comment:
nit: we can avoid memo here

Reply:
You're totally right. I can just use the value directly